benchmark/utils/general.hpp

```diff
@@ -202,6 +248,26 @@ std::vector<std::string> split(const std::string& s, char delimiter = ',')
 }
 
 
+// returns the stream to be used as input of the application
+std::istream& get_input_stream()
+{
+    static auto stream = []() -> std::unique_ptr<std::istream> {
+        std::string input_str(FLAGS_input);
+        if (input_str.empty()) {
+            return nullptr;
+        }
+        if (input_str.back() == ']') {
+            return std::make_unique<std::stringstream>(input_str);
+        }
+        return std::make_unique<std::ifstream>(input_str);
+    }();
+    if (stream) {
+        return *stream;
+    }
```

Comment:
will the data exist when you only take the content of unique_ptr after the unique_ptr is destroyed?

Reply:
the lifetime of the static object ends past the end of the application, so I think we can ignore that possibility.